internal/testutil/kube_api_http_client.go

```diff
@@ -30,22 +31,20 @@ func GetEndpointOnCluster(ctx context.Context, namespace, service, path string,
 		return nil, err
 	}
 
-	req, _ := http.NewRequestWithContext(ctx, http.MethodPost, requestURL.String(), nil) // OK
-	resp, err := client.Do(req)
-	// resp, err := client.Get(requestURL.String())
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, requestURL.String(), nil)
 	if err != nil {
 		return nil, err
 	}
-	defer func() {
-		err = resp.Body.Close()
-	}()
-
-	body, err := io.ReadAll(resp.Body)
+	resp, err := client.Do(req)
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		err = errors.Join(resp.Body.Close())
```

Comment:
this must be 
```suggestion
		err = errors.Join(err, resp.Body.Close())
```

otherwise a potential error in `err` is overridden